### PR TITLE
Bugfix/missing padding auto rotation defaults

### PR DESCRIPTION
**Pull request by pawelfus**
Improved the default padding of axis labels, and fixed their collision calculations. 

Closes #20889

### CODE REVIEW

**Comment by pawelfus** [comment]
Some of the demos now look better, other present the same issue:

1. http://utils.highcharts.local/samples/#test/highcharts/plotoptions/column-pointrange - crowded ticks provide more info, current build looks better
2. http://utils.highcharts.local/samples/#test/highcharts/legend/align - unnecessary rotation, candidate IMHO looks better (applies to all charts with JAN-DEC categories)
3. http://utils.highcharts.local/samples/#test/highcharts/css/error-bar - unable to compare, local build looks the same..

File: ts/Core/Axis/AxisDefaults.ts [hunk 1]
```diff
@@ -1026,11 +1026,12 @@ namespace AxisDefaults {
 
             /**
              * The pixel padding for axis labels, to ensure white space between
-             * them.
+             * them. Defaults to 5 for horizontal axes, 0 for vertical.
```
Comment (TorsteinHonsi) [inline ts/Core/Axis/AxisDefaults.ts:1029]:
```suggestion
             * them. Defaults to 4 for horizontal axes, 1 for vertical.
```

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, this looks good to me! I think all the diff cases now look better with the candidate. Once the docs comment is fixed, are we ready to merge?

**Comment by TorsteinHonsi** [comment]
If this replaces #20889 then we need a proper description and tagging too.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**